sotodlib/io/bookbinder.py

```diff
@@ -212,28 +212,29 @@ def add(self, f):
         """
         self.times.extend(f['data'].times)
 
-        assert f['data'].data.dtype == self.dtypes['data']
         if self.signal is None:
-            self.signal = so3g.G3SuperTimestream(f['data'].names, core.G3VectorTime([]),
-                                    np.empty((len(f['data'].names), 0), dtype=self.dtypes['data']))
-        self.signal.times.extend(f['data'].times)
-        self.signal.data = np.hstack((self.signal.data, f['data'].data))
+            self.signal = {'names': f['data'].names, 'times': [], 'data': []}
+        assert np.asarray(f['data'].names) == np.asarray(self.signal['names']) # there's a bug where G3VectorString(x)==G3VectorString(x) returns False
```

Comment:
Does this work?  Don't you need an all(...)?  I get "ValueError: The truth value of an array with more than one element is ambiguous." for an equivalent test.

Reply:
Oh now I see why my unit test didn’t catch that: it uses a single channel, so it only had one element (and so didn’t have that error). Let me rectify that.